kernel/sched.c

```diff
@@ -798,17 +795,27 @@ void _impl_k_sleep(s32_t duration)
 	/* wait of 0 ms is treated as a 'yield' */
 	if (duration == 0) {
 		k_yield();
-		return;
+		return 0;
 	}
 
 	ticks = _TICK_ALIGN + _ms_to_ticks(duration);
+	expected_wakeup_time = ticks + z_tick_get_32();
```

Comment:
`expected_wakeup_time ` must be unsigned type unless we handle overflow; overflow of signed type is an undefined behavior

Reply:
@pdunaj, Right, and nobody cares. We have https://github.com/zephyrproject-rtos/zephyr/issues/9663 on that.

Reply:
@pfalcon , It is easy to overlook I guess, but this is what the review is for :) Good that there is a ticket to fix it.

Reply:
> It is easy to overlook I guess, but this is what the review is for

Yes, but the problem is deep - all (almost) Zephyt delay-related functions are defined taking s32_t. So, one either has to workaround that everywhere using u32_t/s32_t back-and-forth casts, or just succumb to using s32_t. Neither solution is adequate, and as I told, my impression is that nobody cares ;-).